### ISSUE TITLE
Correct library interface

### ISSUE THREAD

**Issue by chr7** [thombashi/pathvalidate#23]
pathvalidate 2.5.0

In Visual Studio Code, no suggestions are given for functions of pathvalidate, e.g. for `sanitize_filename`:

![image](https://user-images.githubusercontent.com/7242780/161752637-3192b921-8ebe-4931-a91d-38dd4a477fe9.png)

I thought that this is a problem of VSC's pylance therefore I opened [this issue](https://github.com/microsoft/pylance-release/issues/2518). It was closed with the comment that this is a problem how pathvalidate does the import. To fix it, it should be done as described in PEP561 resp. [library interface](https://github.com/python/typing/blob/master/docs/source/libraries.rst#library-interface).

**Comment by thombashi** [comment]
@chr7 
Thank you for your report.

Could you tell me which part of the library violated PEP561?
I thought `pathvalidate` follows `PEP561` which includes `py.typed` file within the library and import `sanitize_filename` at the [top-level module](https://github.com/thombashi/pathvalidate/blob/cff7f351a106e9120be3e015bd38b3f8331bcdd0/pathvalidate/__init__.py#L17).

Furthermore, code suggestions work correctly if I disable `Pylance`:
![image](https://user-images.githubusercontent.com/15517661/162454103-66b158c1-32fd-4da7-bd82-3be8070df3d4.png)

**Comment by chr7** [comment]
I can't tell you this, I forwarded this info from the pylance developer (see [my issue there](https://github.com/microsoft/pylance-release/issues/2518)).

Anyhow, I switched the language server off (was set on pylance previously):

![image](https://user-images.githubusercontent.com/7242780/162936200-b402f4ae-1cfc-4dc6-a2cf-2bd7fa6ed272.png)

and I don't get any suggestions neither which isn't a surprise as the language server is off :-)

![image](https://user-images.githubusercontent.com/7242780/162936390-142897a0-a43d-41c2-9d75-410f12589d13.png)

Then I tried Jedi:

![image](https://user-images.githubusercontent.com/7242780/162937456-8985269f-eb32-43c5-9987-68eee4bdaaa1.png)

I'll pass this info back to pylance, let's see what they say.

**Comment by chr7** [comment]
According to the pylance developer 'Imported symbols are considered private by default. If they use the “import A as A”'
https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface

![image](https://user-images.githubusercontent.com/7242780/163379406-d43c8e64-3b8c-47c9-8f88-983934f04a46.png)

They suggest to fix that.

**Comment by eggplants** [comment]
When I used this library, mypy said:

```
getjump/getjump.py:111: error: Module has no attribute "sanitize_filepath"; maybe "sanitize_file_path",
"sanitize_filename", or "validate_filepath"?  [attr-defined]
            save_dir = pathvalidate.sanitize_filepath(
                       ^
getjump/main.py:91: error: Module has no attribute "sanitize_filepath"; maybe "sanitize_file_path",
"sanitize_filename", or "validate_filepath"?  [attr-defined]
            type=pathvalidate.sanitize_filepath,
                 ^
Found 2 errors in 2 files (checked 3 source files)
```

These errors are maybe related to this issue.

---

With [`__all__`](https://docs.python.org/3/tutorial/modules.html?highlight=__all__#importing-from-a-package), no errors are reported.

```diff
diff --git a/pathvalidate/__init__.py b/pathvalidate/__init__.py
index 63f696c..3b55b9f 100644
--- a/pathvalidate/__init__.py
+++ b/pathvalidate/__init__.py
@@ -35,3 +35,43 @@ from .error import (
     ValidationError,
     ValidReservedNameError,
 )
+
+__all__ = (
+    "__author__",
+    "__copyright__",
+    "__email__",
+    "__license__",
+    "__version__",
+    "AbstractSanitizer",
+    "AbstractValidator",
+    "Platform",
+    "ascii_symbols",
+    "normalize_platform",
+    "replace_ansi_escape",
+    "replace_unprintable_char",
+    "unprintable_ascii_chars",
+    "validate_null_string",
+    "validate_pathtype",
+    "FileNameSanitizer",
+    "is_valid_filename",
+    "sanitize_filename",
+    "validate_filename",
+    "FilePathSanitizer",
+    "is_valid_filepath",
+    "sanitize_file_path",
+    "sanitize_filepath",
+    "validate_file_path",
+    "validate_filepath",
+    "sanitize_ltsv_label",
+    "validate_ltsv_label",
+    "replace_symbol",
+    "validate_symbol",
+    "ErrorReason",
+    "InvalidCharError",
+    "InvalidLengthError",
+    "InvalidReservedNameError",
+    "NullNameError",
+    "ReservedNameError",
+    "ValidationError",
+    "ValidReservedNameError",
+)
```

**Comment by thombashi** [comment]
@eggplants 
Thank you for your report.
Could you tell me how to reproduce the mypy error?
Python/pathvalidate/mypy version, sample code, etc.

**Comment by eggplants** [comment]
```bash
git clone https://github.com/eggplants/getjump
cd getjump
git checkout pathvalidate_error
pip install .
mypy getjump
```
```txt
getjump/getjump.py:113: error: Module has no attribute "sanitize_filepath"; maybe "sanitize_file_path",
"sanitize_filename", or "validate_filepath"?  [attr-defined]
                pathvalidate.sanitize_filepath(
                ^
getjump/main.py:60: error: Module has no attribute "is_valid_filepath"; maybe "validate_filepath",
"is_valid_filename", or "validate_file_path"?  [attr-defined]
        if pathvalidate.is_valid_filepath(v, platform="auto"):
           ^
Found 2 errors in 2 files (checked 3 source files)
```

**Comment by eggplants** [comment]
Or, just:

```shellsession
$ cat a.py
import pathvalidate

res = pathvalidate.sanitize_filepath("hello")
print(res)
$ mypy a.py
a.py:3: error: Module has no attribute "sanitize_filepath"; maybe "sanitize_file_path", "sanitize_filename", or
"validate_filepath"?  [attr-defined]
    res = pathvalidate.sanitize_filepath("hello")
          ^
Found 1 error in 1 file (checked 1 source file)
```

**Comment by thombashi** [comment]
@eggplants 
Thank you for your information. I can reproduce the issue.
Merged the PR and I had released `pathvalidate 2.5.1`.
